src/sql/src/session/vars.rs

```diff
@@ -1317,6 +1328,10 @@ impl SessionVars {
                 &ENABLE_SESSION_CARDINALITY_ESTIMATES,
                 &ENABLE_CARDINALITY_ESTIMATES,
             )
+            .with_value_constrained_var(
+                &MAX_QUERY_RESULT_SIZE,
+                ValueConstraint::Domain(&MaxConstraint(MAX_QUERY_RESULT_SIZE_DEFAULT)),
+            )
```

Comment:
So this won't get updated if we try and increase `max_result_size`? Doesn't this prevent increases to `max_result_size` from having an effect?

Reply:
Hmm that's annoyingly true, and so this probably needs to be taught about that? Ergh.

Reply:
Changed the peek impl to consult both the system and session vars. This will allow the server to be configured by LD, and sessions can opt in to smaller values if they want without needing to plumbing the LD changes through to the session vars.